File: src/contests/cg/puzzle/ruby/teds-compiler.rb
# 🎮 CodinGame Puzzle - teds-compiler
# https://www.codingame.com/training/easy/teds-compiler

line = gets.chomp
balance = 0
best = 0
line.chars.each_with_index do |c, i|
  if c == "<"
    balance += 1
  else
    balance -= 1
  end
  break if balance < 0
  best = i + 1 if balance == 0
end
puts best

File: src/contests/cg/puzzle/scala/teds-compiler.scala
// 🎮 CodinGame Puzzle - teds-compiler
// https://www.codingame.com/training/easy/teds-compiler

import scala.util.control.Breaks._

object Solution {
  def main(args: Array[String]): Unit = {
    val line = scala.io.StdIn.readLine().replaceAll("\\s+$", "")
    var balance = 0
    var best = 0
    breakable {
      for (i <- 0 until line.length) {
        if (line.charAt(i) == '<') balance += 1 else balance -= 1
        if (balance < 0) break()
        if (balance == 0) best = i + 1
      }
    }
    println(best)
  }
}
